File: py/1614.py
class Solution:
    def maxDepth(self, s: str) -> int:
        count = 0
        res = []
        for i in s:
            if i=='(':
                res.append('(')
            elif i==')':
                res.pop()
            count = max(len(res),count)
            
        return count
        

File: cpp/1614.cpp
#include<algorithm>
class Solution {
public:
    int maxDepth(string s) {
        std::stack<char> st;
        int count1 = 0;
        for(int i=0;i<s.size();i++){
            if (s[i]=='('){
                st.push('(');
            }
            else if(s[i]==')'){
                st.pop();
            }
            count1 = count1>st.size()?count1:st.size();
        }
        return count1;
    }
};
